src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -253,21 +279,18 @@ def updatePileup(self, doc, rseList=None, validate=True, userDN=''):
             self.logger.error(err)
             return [err.error()]
 
-        # check previous fraction value and add new transition record if it is
-        # differ from current one or does not exist, see logic:
-        # https://gist.github.com/amaltaro/b4f9bafc0b58c10092a0735c635538b5
-        prevTranRecord = transition[-1]
-        if prevTranRecord['containerFraction'] != fraction:
-            customName = customDID(prevTranRecord['customDID'])
-            # preserve previous container fraction
-            transitionRecord = {'containerFraction': prevTranRecord['containerFraction'],
-                                'customDID': customName,
-                                'updateTime': gmtimeSeconds(),
-                                'DN': userDN}
-            transition.append(transitionRecord)
-            doc['transition'] = transition
-            self.logger.info("Added transition record for pileup %s", pname)
+        # update transition record if necessary
+        updateTransitionRecord(doc, userDN, self.logger)
+        output = self.updatePileupDocumentInDatabase(spec, doc)
+        return output
 
+    def updatePileupDocumentInDatabase(self, spec, doc):
+        """
+        Update given pileup document in database
+        :param spec: MongoDB spec (dictionary) used to locate document in MongoDB
+        :param doc: pileup document (dictionary)
+        :return: outcome of update, either empty list in case of success or list of error
+        """
         # mandatory timestamp updates
         doc.update(getNewTimestamp(doc))
```

Comment:
Unrelated to these changes, but this is a strange call and I see room for refactoring improvements (getNewTimestamp(doc) alone would do it).